canary/producer-cpp/src/CanaryConfig.cpp

```diff
@@ -0,0 +1,75 @@
+#include "CanaryConfig.h"
+
+CanaryConfig::CanaryConfig()
+{
+    testVideoFps = 25;
+    streamName = "DefaultStreamName";
+    sourceType = "TEST_SOURCE";
+    canaryRunScenario = "Continuous"; // (or intermittent)
+    streamType = "REALTIME";
+    canaryLabel = "DEFAULT_CANARY_LABEL"; // need to decide on a default value
+    cpUrl = "";
+    fragmentSize = DEFAULT_FRAGMENT_DURATION_MILLISECONDS;
+    canaryDuration = DEFAULT_CANARY_DURATION_SECONDS;
+    bufferDuration = DEFAULT_BUFFER_DURATION_SECONDS;
+    storageSizeInBytes = 0;
+    useAggMetrics = true;
+}
+
+void CanaryConfig::setEnvVarsString(string &configVar, string envVar)
+{
+    if (getenv(envVar.c_str()) != NULL)
+    {
+        configVar = getenv(envVar.c_str());
+    }
+}
+
+void CanaryConfig::setEnvVarsInt(int &configVar, string envVar)
+{
+    if (getenv(envVar.c_str()) != NULL)
+    {
+        configVar = stoi(getenv(envVar.c_str()));
```

Comment:
For string to int or other data type conversions, reference this: https://github.com/awslabs/amazon-kinesis-video-streams-pic/blob/master/src/utils/include/com/amazonaws/kinesis/video/utils/Include.h. There is a strtoint() function that does this.